satpy/readers/nwcsaf_nc.py

```diff
@@ -256,13 +258,16 @@ def _prepare_variable_for_palette(self, variable, info):
         except KeyError:
             scale = 1
             offset = 0
+            fill_value = 255
```

Comment:
Are we sure that missing scale and offset also calls for setting the fill_value here?

Reply:
Yes, sort of. The fill_value need to come from the same dataset as scale/offset. If no dataset is supplied (cloudmask, cloudtype, cloud phase, or geo palette) we have to use default values. Without fill_value_color in palette attributes (geo) nothing will happen with this value.